### DISCUSSION TITLE
How to indent in a snippet?

### DISCUSSION THREAD

**Discussion by Tgist1024** [ayamir/nvimdots#599] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/101801875/227837827-84a10d56-4ae9-4df7-9101-ce410c9c9cf6.png)
I want to indent here, but the `tab` key cannot work.
Is there any way to indent in a snippet? This is necessary for some indent-sensitive languages.

**Comment by charliie-dev** [comment]
ctrl+>>

**Reply by Tgist1024** [reply]
emm...which one is the `>>` key?

**Reply by charliie-dev** [reply]
just type `>` two times lol

**Reply by Tgist1024** [reply]
just `..` happened.

**Reply by Tgist1024** [reply]
shift + >> works, but only in normal mode, is there method work in insert mode?

**Reply by charliie-dev** [reply]
Sorry! It should be `shift`. Smooth brain here.

**Reply by charliie-dev** [reply]
word motions typically work in normal mode. I'm not sure how to do that in insert mode either.
Maybe u can try this plugin: https://github.com/NMAC427/guess-indent.nvim

**Reply by Tgist1024** [reply]
thanks, I will have a try.

**Comment by Jint-lzxy** [comment, marked as answer]
@Tgist1024 If you want `neovim` to interpret your next input _literally_, you can always use `<C-v>` in insert mode (or `<C-q>` if `<C-v>` is not available). For this situation, you may use the combination `<C-v><Tab>`.

In normal mode, @CharlesChiuGit's solution should suffice.

FYI:
```console
						*i_CTRL-V*
CTRL-V		Insert next non-digit literally.  It's also possible to enter
		the decimal, octal or hexadecimal value of a character
		|i_CTRL-V_digit|.
		The characters typed right after CTRL-V are not considered for
		mapping.
		For special keys, the CTRL modifier may be included into the
		key to produce a control character.  If there is no control
		character for the key then its |key-notation| is inserted.
		Note: When CTRL-V is mapped (e.g., to paste text) you can
		often use CTRL-Q instead |i_CTRL-Q|.

						*i_CTRL-Q*
CTRL-Q		Same as CTRL-V.
		Note: Some terminal connections may eat CTRL-Q, it doesn't
		work then.  It does work in the GUI.
```

**Reply by Jint-lzxy** [reply]
https://github.com/ayamir/nvimdots/pull/611 should fix this.
